imap/idled.c

```diff
@@ -87,170 +114,211 @@ EXPORTED void fatal(const char *msg, int err)
     exit(err);
 }
 
-static int mbox_count_cb(const mbentry_t *mbentry __attribute__((unused)), void *rockp)
+static int alert_cb(sqlite3_stmt *stmt, void *rock)
 {
-    int *ip = (int *) rockp;
-    (*ip)++;
+    json_t *msg = rock;
+    struct sockaddr_un *client =
+        (struct sockaddr_un *) sqlite3_column_blob(stmt, 0);
 
+    idle_send(client, msg);
     return 0;
 }
 
-/* remove an ientry from list of those idling on mboxname */
-static void remove_ientry(const char *mboxname,
-                          const struct sockaddr_un *remote)
+struct notify_rock {
+    json_t *msg;
+    arrayu64_t *failed_pids;
+};
+
+static int notify_cb(sqlite3_stmt *stmt, void *rock)
 {
-    struct ientry *t, *p = NULL;
+    struct notify_rock *nrock = rock;
+    const char *mboxid =
+        json_string_value(json_object_get(nrock->msg, "mailboxID"));
```

Comment:
we should probably make this all lowercase - that's what JSON API users are used to, and maybe there will be pluggable IDLE engines

Reply:
Legit